sound/soc/codecs/rt5682.c

```diff
@@ -2818,6 +2822,7 @@ static int rt5682_register_dai_clks(struct snd_soc_component *component)
 			init.parent_hws = &(const struct clk_hw *){
 				&rt5682->dai_clks_hw[RT5682_DAI_WCLK_IDX]
 			};
+			barrier();
```

Comment:
what is this trying to do?

Reply:
@plbossart 
I found the point where assigns the wrong pointer address in the clk_core_populate_parent_map function.
https://github.com/thesofproject/linux/blob/e1e85cff35eb5c92ab618216c052bb2fb669f307/drivers/clk/clk.c#L3769
And then, the 'general protection fault' issue happened when accessing the clk_core of clk_hw.
https://github.com/thesofproject/linux/blob/e1e85cff35eb5c92ab618216c052bb2fb669f307/drivers/clk/clk.c#L426

The 'parent_hws[i]' come from the codec driver, please check below.
https://github.com/thesofproject/linux/blob/e1e85cff35eb5c92ab618216c052bb2fb669f307/sound/soc/codecs/rt5682.c#L2818
If I try to print the address of 'init.parent_hws[i]' in the codec driver, the issue will not occur.
I feel the compiler optimization might cause this issue.
Therefore, I added the barrier() after the 'init.parent_hws' assignment and the test result is ok.

Reply:
not getting the explanation, sorry @shumingfan